packages/tools/src/tools/segmentation/strategies/fillRectangle.ts

```diff
@@ -57,8 +59,40 @@ function fillRectangle(
     segmentationImageData.getDimensions()
   );
 
-  // Since always all points inside the boundsIJK is inside the rectangle...
-  const pointInRectangle = () => true;
+  const isStackViewport = viewport instanceof StackViewport;
+
+  // Are we working with 2D rectangle in axis aligned viewport view or not
+  const isAligned =
+    isStackViewport || isAxisAlignedRectangle(rectangleCornersIJK);
+
+  const direction = segmentationImageData.getDirection();
+  const spacing = segmentationImageData.getSpacing();
+  const { viewPlaneNormal } = viewport.getCamera();
+
+  // In case that we are working on oblique, our EPS is really the spacing in the
+  // normal direction, since we can't really test each voxel against a 2D rectangle
+  // we need some tolerance in the normal direction.
+  const EPS = csUtils.getSpacingInNormalDirection(
+    {
+      direction,
+      spacing,
+    },
+    viewPlaneNormal
+  );
+
+  const pointsBoundsLPS = getBoundingBoxAroundShape(points);
+  const [[xMin, xMax], [yMin, yMax], [zMin, zMax]] = pointsBoundsLPS;
+
+  const pointInShapeFn = isAligned
+    ? () => true
+    : (pointLPS) => {
+        const [x, y, z] = pointLPS;
+        const xInside = x >= xMin - EPS && x <= xMax + EPS;
```

Comment:
Given how many times this might be called, it is worth doing the +/- EPS before hand by using let for the assignment, and then changing xMin etc